libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1087,10 +1089,18 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	{
 		// No initial value is only permitted for single variables with specified type.
 		if (_statement.declarations().size() != 1 || !_statement.declarations().front())
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			// Error already reported in syntax checker, but we need to bail out here?
+			if (!boost::algorithm::all_of_equal(_statement.declarations(), nullptr))
+				m_errorReporter.fatalTypeError(_statement.location(), "Use of the \"var\" keyword is disallowed.");
+			else
+				return false;
+		}
+
 		VariableDeclaration const& varDecl = *_statement.declarations().front();
 		if (!varDecl.annotation().type)
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+			m_errorReporter.fatalTypeError(_statement.location(), "Use of the \"var\" keyword is disallowed");
```

Comment:
Yes, here the dot is missing. Now we now that the tests catch this branch, though :).